### PR TITLE
STATS-282: Replace summary period pills with the date range picker

### PR DESCRIPTION
**Pull request by Nikschavan**
Part of STATS-282

## Proposed Changes

* Replace the Day / 7 days / 30 days / Quarter / Year / All Time segmented control on stats summary pages with the same date range picker used on the Traffic page.
* Add an "All time" shortcut to the picker (gated like the other shortcuts) that keeps the legacy `num=-1&summarize=1` URL contract.
* Summary pages now accept the modern `chartStart`/`chartEnd` params, falling back to the legacy `startDate`/`endDate` ones so old links keep working.
* Module "View details" links forward the active range/shortcut, and back links/breadcrumbs carry the range back to the Traffic page.

https://github.com/user-attachments/assets/1be77fdb-6935-4f5a-a701-bf3b8ec1ab3a

## Why are these changes being made?

* Summary pages used their own period pills while the Traffic page already has a full date range picker, so date selection behaved differently between the two. Using the same picker everywhere unifies the UX and lets summary pages support arbitrary custom ranges.

## Testing Instructions

1. Go to Stats > Traffic (`/stats/day/:site`), select a date range, then open "View details" on a module (e.g. Posts & pages) — the summary page should open with the same range preselected in the picker instead of the old period pills.
2. On the summary page, switch ranges via the picker (shortcuts and a custom range) and confirm the module data updates and the URL uses `chartStart`/`chartEnd`.
3. Select the "All time" shortcut — the URL should switch to `num=-1&summarize=1`, the label should read "All Time", and on a free site the shortcut should show the upsell instead.
4. Use the back link/breadcrumb — it should return to Traffic with the selected range intact.
5. Open a legacy summary URL (e.g. `?startDate=YYYY-MM-DD&num=30&summarize=1`) and confirm it still renders the correct range.

Before/After screenshots: _(placeholder — to be added by the author)_

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://github.com/Automattic/wp-calypso/blob/trunk/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] For UI changes, have you tested the affected components in [dark mode](https://github.com/Automattic/wp-calypso/blob/trunk/docs/dark-mode.md)?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages.
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/utils.js [hunk 1]
```diff
@@ -39,6 +39,41 @@ export function getPathWithUpdatedQueryString( query = {}, path = page.current )
...
+
+	if ( ! query?.start_date ) {
+		return `${ url }?startDate=${ period.endOf.format( DATE_FORMAT ) }`;
+	}
+
+	// The range and shortcut originate from the URL, so serialize instead of interpolating.
+	const params = new URLSearchParams( {
+		chartStart: query.start_date,
+		chartEnd: query.date,
+	} );
```
[the remark this answers is not in the document]
Comment (Nikschavan) [reply client/my-sites/stats/utils.js]: Fixed in 8c6b26c — `buildSummaryUrl` now falls back to the legacy single-date link unless both `start_date` and `date` are present.

File: client/my-sites/stats/stats-module/all-time-nav.jsx [hunk 2]
```diff
@@ -32,215 +29,108 @@ export const StatsModuleSummaryLinks = ( props ) => {
...
+	const dateRange = { chartStart, chartEnd, daysInRange, shortcutId };
+
+	const fullShortcutList = useMemo(
+		() => [
+			...shortcutList,
+			{
+				id: 'all_time',
+				label: translate( 'All time' ),
+				startDate: '',
+				endDate: '',
```
[the remark this answers is not in the document]
Comment (Nikschavan) [reply client/my-sites/stats/stats-module/all-time-nav.jsx:82]: Fixed in 8c6b26c — restored the "All Time" casing. This also reuses the existing translated string instead of introducing a new one.

File: client/my-sites/stats/features/modules/stats-utm/stats-module-utm.jsx [hunk 3]
```diff
@@ -145,11 +145,15 @@ const StatsModuleUTM = ( {
...
+				// Only fall back to the legacy single-date contract when a modern
+				// `chartStart`/`chartEnd` range isn't already being forwarded.
+				if ( ! clonedParams.has( 'chartStart' ) ) {
+					if ( ! clonedParams.has( 'startDate' ) ) {
+						clonedParams.set( 'startDate', period.startOf.format( 'YYYY-MM-DD' ) );
+					}
+					if ( ! clonedParams.has( 'endDate' ) ) {
+						clonedParams.set( 'endDate', period.endOf.format( 'YYYY-MM-DD' ) );
+					}
 				}
```
[the remark this answers is not in the document]
Comment (Nikschavan) [reply client/my-sites/stats/features/modules/stats-utm/stats-module-utm.jsx]: Fixed in 8c6b26c — a lone `chartStart` or `chartEnd` is now treated as no modern range: both params are dropped and the URL falls back to the legacy `startDate`/`endDate` contract.

File: client/components/stats-date-control/index.tsx [hunk 4]
```diff
@@ -122,9 +133,7 @@ const StatsDateControl = ( {
 		shortcutId?: string
 	) => {
 		const queryParamsObject = qs.parse( queryParams );
-		removeOutOfRangeStartDate( queryParamsObject, startDate, endDate );
-		delete queryParamsObject.shortcut;
-		delete queryParamsObject.navigation;
+		removeLegacyRangeParams( queryParamsObject );
```
[the remark this answers is not in the document]
Comment (Nikschavan) [reply client/components/stats-date-control/index.tsx:135]: I checked this against the `qs` version installed in the repo — object input is an official overload (`parse(str: string | Record<string, string>, ...)` in `@types/qs`), and it returns the parsed params as a new object rather than coercing to "[object Object]". Since it's a new object, the `delete` calls below also can't mutate `context.query`. The Traffic page has been passing `context.query` through this same path since before this PR. I've typed the prop as `string | Record<string, string>` so the contract is explicit.

File: client/components/stats-date-control/index.tsx [hunk 5]
```diff
@@ -138,8 +147,27 @@ const StatsDateControl = ( {
...
-		const url = `/stats/${ period }/${ slug }`;
-		return `${ url }${ newRangeQuery }`;
+		return `${ basePath( period ) }${ newRangeQuery }`;
+	};
+
+	// "All time" keeps the legacy `num=-1&summarize=1` contract so the existing
+	// query/label logic keeps working instead of a synthetic huge chart range.
+	const generateAllTimeLink = () => {
+		const queryParamsObject = qs.parse( queryParams );
+		removeLegacyRangeParams( queryParamsObject );
```
[the remark this answers is not in the document]
Comment (Nikschavan) [reply client/components/stats-date-control/index.tsx:156]: Same as the thread above — `qs.parse` accepts an object per its documented overload and returns a fresh object, so no normalization is needed here either.

**Review by kangzj — APPROVED** [review]
Nice unification of the summary-page date selection with the Traffic picker — the legacy/modern URL contract handling and the extraction into testable pure functions (`getSummaryDateRangeFromQuery`, `buildSummaryUrl`) are solid, and the navigation-history edge cases (no prior history, stale shortcuts, all-time range preservation) are well covered by tests.

Approving, with two non-blocking observations for follow-up:

1. **No direct test coverage for `all-time-nav.jsx` (`StatsModuleSummaryLinks`)** — the component at the center of this PR, rewritten from the segmented control to `StatsDateControl`. None of its derivations (`isAllTime`, the `chartStart`/`chartEnd` seeding branches for legacy pill vs. plain period-end links, `shortcutId` resolution) have unit tests, unlike the rest of the PR. Worth adding a few cases before/after merge.

2. **Analytics event dropped, not replaced 1:1** — the old `recordStats()` fired `recordGoogleEvent('Stats', 'Clicked Summary Link: ...')` and a `stats_summary_nav_period_clicked` Tracks event on every period click. Clicks now only fire the shared `*_date_picker_shortcut_*_clicked` events from `StatsDateControl`. Presumably intentional as part of unifying analytics with the Traffic picker, but worth confirming with whoever owns Stats dashboards that keying off `stats_summary_nav_period_clicked` isn't still expected for summary pages.
